src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,111 @@
-import { Component } from 'react';
-
-type Props = {};
-type State = {};
-
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
-
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+import React, { useState } from 'react';
+
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitileError, setTitileError] = useState(false);
+
+  const [description, setDescription] = useState('');
+
+  const [imgUrl, setImgUrl] = useState('');
+  const [hasImgUrlError, setImgUrlError] = useState(false);
+
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [hasImdbUrlError, setImdbUrlError] = useState(false);
+
+  const [imdbId, setImdbId] = useState('');
+  const [hasImdbIdError, setImdbIdError] = useState(false);
+
+  const clearInputs = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const checkErrors = () => {
+    setTitileError(!title);
+    setImgUrlError(!imgUrl);
+    setImdbUrlError(!imdbUrl);
+    setImdbIdError(!imdbId);
+  };
+
+  const haldleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    checkErrors();
+
+    if (title && imgUrl && imdbUrl && imdbId) {
+      onAdd({
+        title, description, imgUrl, imdbUrl, imdbId,
+      });
+
+      clearInputs();
+    }
+  };
+
+  return (
+    <form
+      onSubmit={haldleSubmit}
+    >
+      <input
+        type="text"
+        placeholder="Title"
+        className={hasTitileError ? 'error' : ''}
+        value={title}
+        onChange={(event) => {
+          setTitle(event.target.value);
+          setTitileError(false);
+        }}
+      />
+
+      <input
+        type="text"
+        placeholder="Description"
+        value={description}
+        onChange={(event) => {
+          setDescription(event.target.value);
+        }}
+      />
+
+      <input
+        type="text"
+        className={hasImgUrlError ? 'error' : ''}
+        placeholder="Image URL"
+        value={imgUrl}
+        onChange={(event) => {
+          setImgUrl(event.target.value);
+          setImgUrlError(false);
+        }}
+      />
+
+      <input
+        type="text"
+        className={hasImdbUrlError ? 'error' : ''}
+        placeholder="IMDB URL"
+        value={imdbUrl}
+        onChange={(event) => {
+          setImdbUrl(event.target.value);
+          setImdbUrlError(false);
+        }}
+      />
+
+      <input
+        type="text"
+        className={hasImdbIdError ? 'error' : ''}
```

Comment:
```suggestion
        className={hasImdbIdError ? 'error' : ''}
```
Для таких випадків можна використовувати бібліотеку classnames
https://www.npmjs.com/package/classnames

Reply:
так я за неї знаю, але були проблеми з інтернетом, то я вже вирішила не ризикувати інталювати бібліотеку) Зазвичай використовую її.